x/commitment/keeper/keeper.go

```diff
@@ -299,3 +309,82 @@ func (k Keeper) ProcessWithdrawValidatorCommission(ctx sdk.Context, delegator st
 
 	return nil
 }
+
+// Withdraw Token - USDC
+// Only withraw USDC from dexRevenue wallet
+func (k Keeper) ProcessWithdrawUSDC(ctx sdk.Context, creator string, denom string, amount sdk.Int) error {
+	if denom != ptypes.USDC {
+		return sdkerrors.Wrapf(types.ErrWithdrawDisabled, "denom: %s", denom)
+	}
+
+	assetProfile, found := k.apKeeper.GetEntry(ctx, denom)
+	if !found {
+		return sdkerrors.Wrapf(aptypes.ErrAssetProfileNotFound, "denom: %s", denom)
+	}
+
+	if !assetProfile.WithdrawEnabled {
+		return sdkerrors.Wrapf(types.ErrWithdrawDisabled, "denom: %s", denom)
+	}
+
+	commitments, err := k.DeductCommitments(ctx, creator, denom, amount)
+	if err != nil {
+		return err
+	}
+
+	// Update the commitments
+	k.SetCommitments(ctx, commitments)
+
+	// Emit Hook commitment changed
+	k.AfterCommitmentChange(ctx, creator, sdk.NewCoin(denom, amount))
+
+	// Emit blockchain event
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeCommitmentChanged,
+			sdk.NewAttribute(types.AttributeCreator, creator),
+			sdk.NewAttribute(types.AttributeAmount, amount.String()),
+			sdk.NewAttribute(types.AttributeDenom, denom),
+		),
+	)
+
+	return nil
+}
+
+// Withdraw validator's USDC commission to self delegator
+func (k Keeper) ProcessWithdrawValidatorCommissionUSDC(ctx sdk.Context, delegator string, creator string, denom string, amount sdk.Int) error {
```

Comment:
Commitment module should handle validator logic as well?

Reply:
This is also handling the uncommitted token amount in commitment module. But let me try to use the same function for validators as well.

Reply:
Okay